src/ORM/DataList.php

```diff
@@ -1458,8 +1517,29 @@ private function fillEmptyEagerLoadedRelations(
     public function eagerLoad(...$relationChains): static
     {
         $list = clone $this;
-        foreach ($relationChains as $relationChain) {
-            // Don't add any relations we've added before
+
+        // If an array is passed in directly, treat it as though $relationChains wasn't spread.
+        if (count($relationChains) === 1 && is_array($relationChains[array_key_first($relationChains)])) {
+            $relationChains = $relationChains[array_key_first($relationChains)];
+        }
```

Comment:
Handles the difference between passing strings (original API, useful for templates) and passing an array (enhanced API, necessary if applying manipulations)